libs/execution_context/ebpf_core.c

```diff
@@ -365,6 +367,27 @@ _ebpf_core_protocol_create_program(
     return retval;
 }
 
+static bool
+_ebpf_is_find_update_delete_operation_permitted_on_map(_In_ const ebpf_map_definition_t* map_definition)
```

Comment:
The precedent we have been using is that the function pointer is null if not permitted, so shouldn't need a separate bool check.  See libs/execution_context/ebpf_maps.c  line 893 for example.   Also, map-specific code should be in ebpf_maps.c not ebpf_core.c